notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,151 @@
+
+  upmodal_notice.init = true;
+
+  (function() {
+
+    var defaults = {
+      selectors: {
+        content:         '.modal-content',
+        title:           '.title',
+        body:            '.body',
+        link:            '.link',
+        actions:         '.notification-actions',
+        actionTemplate:  '.action-template'
+      },
+      readyCallback: function() {}
+    };
+
+    // First 'prime-the-pump' with an ActionURL
+    function initNotices($, settings, callback) {
+      $.ajax({
+        type: 'POST',
+        url: settings.invokeNotificationServiceUrl,
+        async: false,
+        success: function() {
+          fetchNotices($, settings, callback);
+        }
+      });
+    }
+
+    // Then fetch the notifications with a ResourceURL
+    function fetchNotices($, settings, callback) {
+      $.ajax({
+        url      : settings.getNotificationsUrl,
+        type     : 'POST',
+        dataType : 'json',
+        success: function (data) {
+          feed = data.feed;
+          callback(feed);
+        },
+        error: function () {
+          container.html(" ").text("AJAX failed. ~ THE END ~");
+        }
+      });
+
+    }
+
+    upmodal_notice.show = function ($, container, options) {
+
+      var settings = $.extend({}, defaults, options);
+
+      var drawActions = function(actionsContainer, alert) {
+
+          var availableActions = alert.availableActions;
+          var actionTemplate = actionsContainer.find(settings.selectors.actionTemplate);
+
+          for (var i=0; i < availableActions.length; i++) {
+              var action = availableActions[i];
+
+              var actionUrl = settings.invokeActionUrlTemplate
+                      .replace('NOTIFICATIONID', alert.id)
+                      .replace('ACTIONID', action.id);
+
+              var actionElement = actionTemplate.clone();
+              actionElement.removeClass('action-template');
+              actionElement.toggleClass('hidden');
+              actionElement.addClass('action');
+              actionElement.find('a').attr('href', actionUrl).html("ACCEPT");
+              actionElement.appendTo(actionsContainer);
+          }
+
+          actionsContainer.toggleClass('hidden');
+
+      }
+
+      var drawNotices = function(feed) {
+
+        // Do we have any notices to show?
+        if (feed && feed.length != 0) {
+
+          // Iterate the notices
+          var len = feed.length;
+          for (var i=0; i < len; i++) {
+            var alert = feed[i];
+
+            var element = container.find(settings.selectors.content);
+
+            // Insert context
+            element.find(settings.selectors.title).html(alert.title);
+            if (alert.body) {
+              element.find(settings.selectors.body).html(alert.body);
+            }
+
+            if (alert.url) {
+              var linkText = alert.linkText || alert.url;
+              element.find(settings.selectors.link).attr('href', alert.url).html(linkText);
```

Comment:
:question: should this be a separate link, rather than rewriting an existing link?

Reply:
Nope